pkg/restapi/vc/operation/operations.go

```diff
@@ -693,11 +693,32 @@ func (o *Operation) retrieveCredentialHandler(rw http.ResponseWriter, req *http.
 		return
 	}
 
-	docURLs, err := o.queryVault(profile, id)
-	if err != nil {
-		o.writeErrorResponse(rw, http.StatusInternalServerError, err.Error())
+	var docURLs []string
 
-		return
+	backoff := 250 * time.Millisecond
```

Comment:
These values (initial backoff, maxRetries, and backoff factor) should be configurable.

Reply:
you should add retry logic outside operation as common library in edge-core

Reply:
Done - and it's being used here now.